lib/Frontend/Frontend.cpp

```diff
@@ -52,8 +53,10 @@ std::string CompilerInvocation::getPCHHash() const {
   return llvm::APInt(64, Code).toString(36, /*Signed=*/false);
 }
 
-void CompilerInstance::createSILModule(bool WholeModule) {
+void CompilerInstance::createSILModule() {
   assert(MainModule && "main module not created yet");
+  // Assume WMO if a -primary-file option was not provided.
+  bool WholeModule = !Invocation.getFrontendOptions().PrimaryInput.hasValue();
```

Comment:
Does this happen far enough along in the process to use your new `isWholeModuleCompilation` instead?

Reply:
Yup! I hadn't been sure the tests were equivalent, since one uses PrimaryInput and the other uses PrimaryBufferID. But I've made the fix.

Reply:
Now I see I had added a comment about PrimaryBufferID and PrimaryInput before, so I must have just missed this one.